src/ws/cockpitsshtransport.c

```diff
@@ -513,6 +548,241 @@ cockpit_ssh_data_free (CockpitSshData *data)
   g_free (data);
 }
 
+
+/* ----------------------------------------------------------------------------
```

Comment:
Is there a reason this code can't live in a separate file? This file is pretty big already. If it's too hard, that's fine, your call.